### PR TITLE
Added ability to do a CONNECT request through the proxy

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/lantern-internal#1100

Depends on new project https://github.com/getlantern/preconn

### CODE REVIEW

File: proxy_http.go [hunk 1]
```diff
@@ -200,8 +213,6 @@ func prepareRequest(req *http.Request) *http.Request {
 	req.URL.Scheme = "http"
 	// We need to make sure the host is defined in the URL (not the actual URI)
 	req.URL.Host = req.Host
-	req.URL.RawQuery = req.URL.RawQuery
```
Comment (atavism) [inline proxy_http.go]: Why do we no longer copy the `RawQuery` here anymore? Is it no longer needed?

Comment (oxtoacart) [reply proxy_http.go]: It was just something the linter complained about. I don't think we need to copy it for any reason.

Comment (atavism) [reply proxy_http.go]: Ok, thanks! This LGTM otherwise.

**Comment by oxtoacart** [comment]
@atavism Thanks for the review!

**Pull request merged**